modin/data_management/factories/factories.py

```diff
@@ -11,43 +11,90 @@
 # ANY KIND, either express or implied. See the License for the specific language
 # governing permissions and limitations under the License.
 
+"""
+Module contains Factories for all of the supported Modin backends.
+
+Factory is a bridge between calls of IO function from high-level API and its
+actual implementation in the engine, bound to that factory. Each backend is represented
+with a Factory class.
+"""
+
 import warnings
 import typing
 import re
 
 from modin.config import Engine
 from modin.engines.base.io import BaseIO
+from pandas.util._decorators import doc
 
 import pandas
 
+_doc_abstract_factory_class = """
+Abstract {role} factory which allows to override the IO module easily.
+
+This class is responsible for dispatching calls of IO-functions to its
+actual engine-specific implementations.
+"""
+
+_doc_factory_class = """
+Factory of {engine_name} engine.
+
+This class is responsible for dispatching calls of IO-functions to its
+actual engine-specific implementations.
+"""
+
+_doc_factory_prepare_method = """
+Initialize Factory.
+
+Fills in `.io_cls` class attribute with {engine_name} engine lazily.
+"""
+
+
 types_dictionary = {"pandas": {"category": pandas.CategoricalDtype}}
 
 
 class FactoryInfo(typing.NamedTuple):
+    """
+    Structure that stores information about factory.
+
+    Parameters
+    ----------
+    engine : str
+        Name of underlying execution engine.
+    partition : str
+        Name of the partition format.
+    experimental : bool
+        Whether underlying engine is experimental-only.
+    """
+
     engine: str
     partition: str
     experimental: bool
 
 
 class NotRealFactory(Exception):
-    pass
-
-
-class BaseFactory(object):
     """
-    Abstract factory which allows to override the io module easily.
+    ``NotRealFactory`` exception class.
+
+    Raise when no matching factory could be found.
     """
 
+    pass
+
+
+@doc(_doc_abstract_factory_class, role="")
+class BaseFactory(object):  # noqa: D101
```

Comment:
`D101: missing docstring for public class`, we're inheriting docs here, doc-checker script can't handle that case

Reply:
It will be starting from https://github.com/modin-project/modin/pull/3000, but let's leave this for now.

Reply:
Oh, it just came to my mind that I'm talking about numpydoc validation, not via pydocstyle.
Pydocstyle won't handle this, yeah.